File: riposte-microservice-template-core-code/src/test/java/com/myorg/ripostemicroservicetemplate/endpoints/HealthCheckEndpointTest.java
package com.myorg.ripostemicroservicetemplate.endpoints;

import com.nike.riposte.server.http.RequestInfo;
import com.nike.riposte.server.http.ResponseInfo;

import org.junit.Test;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Executor;

import io.netty.channel.ChannelHandlerContext;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

/**
 * Verifies the functionality of {@link HealthCheckEndpoint}
 *
 * @author Nic Munroe
 */
public class HealthCheckEndpointTest {

    private HealthCheckEndpoint healthCheckEndpoint = new HealthCheckEndpoint();

    @Test
    public void healthCheckEndpoint_should_match_all_http_methods() {
        // expect
        assertThat(healthCheckEndpoint.requestMatcher().isMatchAllMethods()).isTrue();
    }

    @Test
    public void healthCheckEndpoint_should_always_return_http_status_code_200() {
        // given
        @SuppressWarnings("unchecked")
        RequestInfo<Void> requestMock = mock(RequestInfo.class);
        Executor executorMock = mock(Executor.class);
        ChannelHandlerContext ctxMock = mock(ChannelHandlerContext.class);

        // when
        CompletableFuture<ResponseInfo<Void>> responseFuture = healthCheckEndpoint.execute(
            requestMock, executorMock, ctxMock
        );
        ResponseInfo<Void> responseInfo = responseFuture.join();

        // then
        assertThat(responseInfo.getHttpStatusCode()).isEqualTo(200);
    }

}

File: riposte-microservice-template-core-code/src/test/kotlin/com/myorg/ripostemicroservicetemplate/endpoints/HealthCheckEndpointTest.kt
package com.myorg.ripostemicroservicetemplate.endpoints

import com.nhaarman.mockitokotlin2.mock
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * Verifies the functionality of [HealthCheckEndpoint]
 */
class HealthCheckEndpointTest {

    private val healthCheckEndpoint = HealthCheckEndpoint()

    @Test
    fun healthCheckEndpoint_should_match_all_http_methods() {
        // expect
        assertThat(healthCheckEndpoint.requestMatcher().isMatchAllMethods).isTrue
    }

    @Test
    fun healthCheckEndpoint_should_always_return_http_status_code_200() {
        // when
        val responseFuture = healthCheckEndpoint.execute(mock(), mock(), mock())
        val responseInfo = responseFuture.join()

        // then
        assertThat(responseInfo.httpStatusCode).isEqualTo(200)
    }
}
